app/config/windows.js

```diff
@@ -8,11 +9,28 @@ const cfg = new Config({
   }
 });
 
+function validateAndFixWindowPosition(position, size) {
+  const displays = electron.screen.getAllDisplays();
+  const [x, y] = position;
+  const positionIsValid = displays.some(({workArea}) => {
+    return x >= workArea.x && x <= workArea.x + workArea.width && y >= workArea.y && y <= workArea.y + workArea.height;
+  });
+
+  if (!positionIsValid) {
+    const {workArea} = electron.screen.getPrimaryDisplay();
+    position[0] = workArea.x + (workArea.width - size[0]) / 2;
+    position[1] = workArea.y + (workArea.height - size[1]) / 2;
+    cfg.set('windowPosition', position);
```

Comment:
We don't need to calculate a valid position. I think that fallback to default position/size is ok.

Reply:
The defaults are private to the [windows config](https://github.com/zeit/hyper/blob/5d0c6355e12ec8d9a89182556fa73227374b488f/app/config/windows.js#L4-L9) module. If I refactor my change to be in `index.js` how would I go about invalidating the set values?

Reply:
Thank you to have fixed this